src/resources/views/crud/fields/relationship/fetch.blade.php

```diff
@@ -127,49 +133,6 @@
 <script>
     // if nullable, make sure the Clear button uses the translated string
     document.styleSheets[0].addRule('.select2-selection__clear::after','content:  "{{ trans('backpack::crud.clear') }}";');
-
-    // if this function is not already on page, for example in fetch_create we add it.
-    // this function is responsible for query the ajax endpoint and fetch a default entry
-    // in case the field does not allow null
-    if (!window.fetchDefaultEntry) {
-        var fetchDefaultEntry = function (element) {
-            var $fetchUrl = element.attr('data-data-source');
-            var $relatedAttribute = element.attr('data-field-attribute');
-            var $relatedKeyName = element.attr('data-connected-entity-key-name');
-            var $return = {};
-
-            return new Promise(function (resolve, reject) {
-                $.ajax({
-                    url: $fetchUrl,
-                    data: {
-                        'q': ''
-                    },
-                    type: 'POST',
-                    success: function (result) {
-                        // if data is available here it means a paginated collection has been returned.
-                // we want only the first to be default.
-                if (typeof result.data !== "undefined"){
-                    $key = result.data[0][$relatedKeyName];
-                    $value = processItemText(result.data[0], $relatedAttribute);
-                }else{
-                    $key = result[0][$relatedKeyName];
-                    $value = processItemText(result[0], $relatedAttribute);
-                }
-
-                $return[$relatedKeyName] = $key;
-                $return[$relatedAttribute] = $value;
-
-                $(element).attr('data-current-value', JSON.stringify($return));
-                resolve($return);
-                    },
-                    error: function (result) {
-                        reject(result);
-                    }
-                });
-            });
-        };
-    }
-
```

Comment:
Hmm.... why is this no longer needed, I wonder?

Reply:
This is only used in: (master here) 
![image](https://user-images.githubusercontent.com/7188159/141127007-ac5e7198-c0dc-4192-bfc6-9e45f7a1041c.png)

But even there, it introduces an inconsistency with other fields (like fetch, select_from_ajax etc) so I submitted a commit to also remove it from `fetch_and_create`.

Reply:
Oh ok. So let me rephrase this:
- Previously, on `master`, to show the selected entry we made an AJAX request to get its info (in order to show the Name not the Id, for example);
- Now, on `repeatable-as-array`, we determine the selected entry using PHP, not JS, so no need to make that AJAX request;

Is this correct?

If so, what are the benefits to making this change? Didn't we decide to populate the initial value with JS for a reason? I think so... I can't remember the reason, but... I think we had a good reason, right?

Reply:
@tabacitu two different things here.

A) get the selected options inside repeatable
B) fetch default entry when field does not allow null. 

This one is specifically to **B)** because old me thought it was a good idea to have the `fetch_and_create` and the `fetch` fields have the "similar behavior" as a normal html5 select element, that is, if it does NOT allow null the first entry is selected by default. That is not what happens in select2 selects, and since this is a select2 select it's and inconsistency so should be removed.

Related to **A)** yes, we were only adding the value as JS because of the repeatable, those changes were made in the field to support repeatable. Since we are making repeatable array with major benefits there is no sense in keeping supporting this, this would be regular select2 fields with desired functionalities `fetch` or `fetch_or_create` (that I am thinking in merging into a single `fetch_and_create` but later we talk about it).